lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -133,6 +134,18 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   auto instrumentableFunctions =
       PIXPassHelpers::GetAllInstrumentableFunctions(*m_DM);
 
+  // Make a vector of the instructions since we may delete some of them as we go
```

Comment:
There's a way to iterate safely while deleting:
```
  for (auto *F : instrumentableFunctions) {
    for (auto &block : F->getBasicBlockList()) {
      for (auto it = block.begin(); it != block.end();) {
        llvm::Instruction &I = *(it++);
        SplitVectorStores(m_DM->GetOP(), I);
      }
    }
  }
```

Reply:
Thanks. I guess both approaches implicitly rely on the fact that the container doesn't reallocate elements...